### PR TITLE
Add global config option to select default SSL provider

### PR DESCRIPTION
**Pull request by cshoredaniel**
Adds global option to select e.g. polarssl, mbedtls, or openssl, or nossl as default when there is a choice of ssl providers for a package (which can be overridden by per-package selection).

This is modified based on comments on previous version of this PR.

There are corresponding patches for packages feed and luci feed that I will be submitting that make this more useful (particularly the luci one).

### CODE REVIEW

**Comment by cshoredaniel** [comment]
Updated; any comments?

**Comment by Ansuel** [comment]
I use lede with this commit and all work well, so there is no problem to use it

**Comment by cshoredaniel** [comment]
Any comments?

**Comment by cshoredaniel** [comment]
Also, this has already been accepted into OpenWrt; can we get LEDE and OpenWrt in sync so that we can use this symbol in packages and luci feeds?

File: package/network/services/hostapd/Config.in [hunk 1]
```diff
@@ -9,15 +9,15 @@ config WPA_SUPPLICANT_NO_TIMESTAMP_CHECK
 
 choice
 	prompt "Choose TLS provider"
-	default WPA_SUPPLICANT_INTERNAL
+	default WPA_SUPPLICANT_INTERNAL if !(SSL_DEFAULT_OPENSSL  || SSL_DEFAULT_FALLBACK_OPENSSL)
+	default WPA_SUPPLICANT_OPENSSL if !(SSL_DEFAULT_OPENSSL || SSL_DEFAULT_FALLBACK_OPENSSSL)
```
Comment (nbd168) [inline package/network/services/hostapd/Config.in]: Copy&paste bug here,  the '!' needs to be removed

**Comment by cshoredaniel** [comment]
Updated

**Comment by cshoredaniel** [comment]
This is fixed copy&paste error mentioned.

**Comment by cshoredaniel** [comment]
Apparently I was looking at the wrong branch when I thought it was in openwrt already.  Sorry.

**Comment by cshoredaniel** [comment]
Updated and tested two or three weeks ago.

**Comment by cshoredaniel** [comment]
Reverified on ar71xx netgear wndr3800 last night, as things went a little haywire for a bit.

**Comment by cshoredaniel** [comment]
Deprecated

**Pull request closed without merging**